src/Renci.SshNet.Tests/Classes/Common/CountdownEventTest.cs

```diff
@@ -189,14 +192,19 @@ public void Wait_ShouldReturnFalseWhenTimeoutExpiresBeforeCountdownEventIsSet()
                 threads[i].Start();
             }
 
-            var start = DateTime.Now;
+            var watch = new Stopwatch();
+            watch.Start();
             var actual = countdownEvent.Wait(timeout);
-            var elapsedTime = DateTime.Now - start;
+            watch.Stop();
 
             Assert.IsFalse(actual);
             Assert.IsFalse(countdownEvent.IsSet);
             Assert.IsFalse(countdownEvent.WaitHandle.WaitOne(0));
-            Assert.IsTrue(elapsedTime >= timeout);
+            // Use precision because it fail in CI pipeline on .NET 7 and also on .NET Framework with Stopwatch.
+            // Locally it works without precision. This is internal .NET implementation, so this is 
+            // not super important
+            var precision = TimeSpan.FromMilliseconds(10);
+            Assert.IsTrue(watch.Elapsed >= timeout.Subtract(precision));
```

Comment:
This is not acceptable, it's faking the timer to get to a desired result.

This entire testcase is wrong and prone to race conditions. It starts up to 20 threads, and then starts measuring time. But it fails to account for the time it takes to start the threads; by the time the last thread is started, the first one might have already completed (or be almost complete), so the 30ms timeout may in fact be a sufficient delay for the event to be signaled. The 20 thread execution will also depend on whether this runs on a 4/8/16/32 core machine - the 32-core machine can start the 20 threads much faster than the 4-core one.

Also, what exactly is this test testing? I would just throw it away if it's just testing .Net intrinsics.

(I added this comment on a previous commit but it's hidden now as I didn't started a review)

Reply:
@zybexXL Thanks for the review. I leave this code because I wanted to show why I change this but I agree that this test is not really useful, it just tests .NET's feature. So maybe I just remove this one Assert? Or do you want to remove the entire file `CountdownEventTest`? 

Similar situation is here: https://github.com/sshnet/SSH.NET/pull/1109/files/79123fbadd40f8758b27c71b66896f51fe7bc8fc#diff-791204553618e1fe9731d44416970367c234ce1e1f6b9f5b12c883a2646d992aR338-R342